src/VisualStudio/LiveShare/Impl/Client/RemoteLanguageServiceWorkspace.cs

```diff
@@ -34,22 +37,31 @@ namespace Microsoft.VisualStudio.LanguageServices.LiveShare.Client
     /// A Roslyn workspace that contains projects that exist on a remote machine.
     /// </summary>
     [Export(typeof(RemoteLanguageServiceWorkspace))]
-    internal sealed class RemoteLanguageServiceWorkspace : Workspace, IDisposable, IRunningDocumentTableEventListener
+    internal sealed class RemoteLanguageServiceWorkspace : CodeAnalysis.Workspace, IDisposable, IRunningDocumentTableEventListener
     {
         private readonly IServiceProvider _serviceProvider;
-        private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactoryService;
         private readonly IThreadingContext _threadingContext;
         private readonly RunningDocumentTableEventTracker _runningDocumentTableEventTracker;
+        private readonly IVsFolderWorkspaceService _vsFolderWorkspaceService;
 
         private const string ExternalProjectName = "ExternalDocuments";
 
         // A collection of opened documents in RDT, indexed by the moniker of the document.
         private ImmutableDictionary<string, DocumentId> _openedDocs = ImmutableDictionary<string, DocumentId>.Empty;
 
-        private CollaborationSession _session;
-        private string _remoteRootPath;
-        private string _externalPath;
-        private RemoteDiagnosticListTable _remoteDiagnosticListTable;
+        private CollaborationSession? _session;
+
+        /// <summary>
+        /// Stores the current base folder path(s) on the client that hold files retrieved from the host workspace(s).
+        /// </summary>
+        private ImmutableHashSet<string> _remoteWorkspaceRootPaths;
+
+        /// <summary>
+        /// Stores the current base folder path on the client that holds registered external files.
+        /// </summary>
+        private ImmutableHashSet<string> _registeredExternalPaths;
```

Comment:
What's the difference between these?

Reply:
One stores files from the host that are in the host's workspace / sln.  The other stores files that are opened on the host, but not necessarily part of their solution (just random other misc files).